### PR TITLE
PSDP: Woodgate algorithm

### PR DESCRIPTION
**Pull request by banrovegrie**
This is wrt #98.

I am working within a subfolder, once done I will delete the folder and make everything streamlined. Right now, the code within `PSDP` isn't inheriting or using anything outside of the subfolder.

### CODE REVIEW

**Comment by banrovegrie** [comment]
The added code shows the rough structure of the algorithm. I finally understood the proof and why steepest descent doesn't work.

What's left?
- How to solve for D_i^* and \bar{D_i} from (26, 27)?
image

- Best way to find the minimizer for f(E_i - \omega_i D_i).
image

**Comment by banrovegrie** [comment]
@FanwangM, I think we can meet on Tuesday this week and the implementation will be complete by then 😛

**Comment by FanwangM** [comment]
Thanks for the hard work on this. 

A few comments on this. 

1. Using your master branch to add new features is not a good practice as we will lose track of the upstream and the git history will not align with the upstream. Therefore, we often create a new branch for adding new codes and we will update the master branch once the upstream is updated.
2. You can of course do experiments with a folder, which is exactly the way of understanding the algorithm. But we don't put these into the pull request, as this makes the git history dirty. You can find examples of codes in https://github.com/theochem/procrustes/blob/master/procrustes/symmetric.py
3. I would guess that `input.py` functions as a test, but we don't put test files like this. An example can be found at https://github.com/theochem/procrustes/blob/master/procrustes/test/test_symmetric.py and you should create a new test file named `test_psdp.py` in the `test` folder.

Please refer to the code review for other comments.

File: procrustes/PSDP/PSDP.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
+import numpy as np
```
Comment (FanwangM) [inline procrustes/PSDP/PSDP.py:1]: Missing header section.

**Comment by FanwangM** [comment]
> @FanwangM, I think we can meet on Tuesday this week and the implementation will be complete by then 😛

Can you send us an email every Monday morning following the email thread we have been using to inform us if you will need to meet with professors or just GSoC mentors? This can help the professors arrange the time in a more timely manner. Thank you.

**Comment by FanwangM** [comment]
Hi Arjo. Can you upload your new codes to GitHub so that we can see the latest changes? Thanks.

**Comment by banrovegrie** [comment]
Hey, I will close this PR and open a new one. The algorithm is complete. Solving eq 26 and 27 were much easier than I thought. Also, as @Ali-Tehrani mentioned I used `scipy.optimize` to find the minimizer for $f(E_i - \omega_i D_i)$.

**Pull request closed without merging**